### PR TITLE
#160609514 Reset password by sending an email link

### PR DESCRIPTION
**Pull request by RonKbS**
#### What does this PR do?
Allow users to reset the password of their accounts.

#### Description of Task to be completed?
Currently, a user is unable to resent their password using the frontend of the application.

This PR allows a user to reset the password for their account by inputting the email. They then get a link via that email which they will be able to use to update the password.

#### How should this be manually tested?

Run the application using `npm start`
Click on `Forgot Password?` on the login page
Submit the `username` and `email` of a registered user.
A notification will pop up for checking the email address.
Using the link in the address, set a new password.
Use the new password details to login.

#### What are the relevant pivotal tracker stories?

[#160609514](https://www.pivotaltracker.com/story/show/160609514)

#### Any background context you want to add?
N/A

## Checklist: I completed these to help reviewers

- [x] My IDE is configured to follow the [**ESlint Conventions**](https://github.com/eslint/eslint)

  No? Unsure? -> configure your IDE, format the code and add the changes with `git add . && git commit --amend`

- [x] I have **added tests** to cover my changes.

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the develop branch.

  No? Unsure? -> execute command `git pull --rebase origin develop`

### CODE REVIEW

File: src/actions/userActions.js [hunk 1]
```diff
@@ -42,6 +46,7 @@ export const fetchUsers = (postData) => dispatch => {
 
 export const loginUser = payload => async dispatch => {
   toast.dismiss();
+  // debugger;
```
Comment (MuhanguziDavid) [inline src/actions/userActions.js]: @RonKbS , please remove the commented out code

File: src/config/axiosInstance.js [hunk 2]
```diff
@@ -19,4 +19,5 @@ export const axiosInstance = axios.create({
   headers,
 });
 
+
```
Comment (MuhanguziDavid) [inline src/config/axiosInstance.js]: The extra space may not be needed

File: src/reducers/passwordResetReducers.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
+import {
```
Comment (MuhanguziDavid) [inline src/reducers/passwordResetReducers.js]: Password reset reducers could be part of the userReducers

File: src/tests/components/ResetPassword.test.js [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+  const getEvent = (name = '', value = '') => ({
+    preventDefault: jest.fn(),
+    target: {
+      name,
+      value,
+    },
+  });
+
+
```
Comment (MuhanguziDavid) [inline src/tests/components/ResetPassword.test.js:31]: The extra line could be removed

**Review by MuhanguziDavid — APPROVED** [review]
LGTM

**Review by samuelbwambale — APPROVED** [review]
@RonKbS I like the way the form is separated from the container component. LGTM.

**Review by kakaemma — APPROVED** [review]
LGTM

**Pull request merged**